firebase.json

```diff
@@ -4,11 +4,20 @@
     "cleanUrls": true,
     "ignore": ["firebase.json", "**/.*", "**/node_modules/**"]
   },
-  "functions": {
-    "predeploy": ["yarn build:functions"],
-    "source": "functions",
-    "runtime": "nodejs18"
-  },
+  "functions": [
+    {
+      "predeploy": ["yarn build:functions"],
+      "source": "functions",
+      "codebase": "maple",
+      "runtime": "nodejs18"
+    },
+    {
+      "predeploy": [". llm/venv/bin/activate && python3.10 -m pip install -r llm/requirements.txt"],
```

Comment:
IIUC, this file is mainly used when you type `firebase deploy ...` so this pre-deploy will not happen inside a Docker environment